assets/js/blocks/reviews-by-product/editor-block.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getOrderArgs, getReviews } from './utils';
+import LoadMoreButton from '../../frontend/components/load-more-button';
+import OrderBySelect from '../../frontend/components/order-by-select';
+import Reviews from '../../frontend/components/reviews';
+import withComponentId from '../../frontend/hocs/with-component-id';
+
+/**
+ * Block rendered in the editor.
+ */
+class EditorBlock extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			reviews: [],
+			totalReviews: 0,
+		};
+
+		this.debouncedLoadFirstReviews = debounce( this.loadFirstReviews.bind( this ), 400 );
+	}
+
+	componentDidMount() {
+		this.loadFirstReviews();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if (
+			prevProps.attributes.orderby !== this.props.attributes.orderby ||
+			prevProps.attributes.productId !== this.props.attributes.productId ||
+      prevProps.attributes.reviewsOnPageLoad !== this.props.attributes.reviewsOnPageLoad
+		) {
+			this.debouncedLoadFirstReviews();
+		}
+	}
+
+	getDefaultArgs() {
+		const { attributes } = this.props;
+		const { order, orderby } = getOrderArgs( attributes.orderby );
+		const { productId, reviewsOnPageLoad } = attributes;
+
+		return {
+			order,
+			orderby,
+			per_page: reviewsOnPageLoad,
+			product_id: productId,
+		};
+	}
+
+	loadFirstReviews() {
+		const args = this.getDefaultArgs();
+
+		getReviews( args ).then( ( { reviews, totalReviews } ) => {
```

Comment:
probably could just call `this.getDefaultArgs()` directly in here and avoid the unnecessary setting of a variable.

Reply:
Done in f6a206626342178161202b36faf146363f32599d.